server/server.go

```diff
@@ -2083,6 +2083,11 @@ func (s *Server) Start() {
 
 	defer s.Noticef("Server is ready")
 
+	// Check is this may be a restart.
+	if s.startupComplete == nil {
```

Comment:
I think one should ask if we want to support calling s.Start() after a s.Shutdown(). If yes, then I think it is a bit more than that, that needs to be done. I believe that there are others things that are created/setup in NewServer() whose state may be changed during Shutdown() and that Start() will not take into account.